src/main/java/gift/service/option/OptionService.java

```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Transactional
+    public void addOptionToGift(Long giftId, OptionRequest optionRequest) {
+        Gift gift = giftRepository.findById(giftId)
+                .orElseThrow(() -> new NoSuchElementException("해당 상품을 찾을 수 없습니다 id :  " + giftId));
+        Option option = optionRequest.toEntity();
+
+        checkDuplicateOptionName(gift, option.getName());
+
+        gift.addOption(option);
+        giftRepository.save(gift);
+    }
+
+    @Transactional(readOnly = true)
+    public List<OptionResponse> getAllOptions() {
+        List<OptionResponse> options = optionRepository.findAll().stream()
+                .map(OptionResponse::from)
+                .toList();
+        return options;
+    }
+
+    @Transactional(readOnly = true)
+    public List<OptionResponse> getOptionsByGiftId(Long giftId) {
+        Gift gift = giftRepository.findById(giftId)
+                .orElseThrow(() -> new NoSuchElementException("해당 상품을 찾을 수 없습니다 id :  " + giftId));
+
+        return gift.getOptions().stream()
+                .map(OptionResponse::from)
+                .toList();
+    }
+
+    @Transactional
+    public void updateOptionToGift(Long giftId, Long optionId, OptionRequest optionRequest) {
+        Gift gift = giftRepository.findById(giftId)
+                .orElseThrow(() -> new NoSuchElementException("해당 상품을 찾을 수 없습니다 id :  " + giftId));
+        Option option = optionRepository.findById(optionId)
+                .orElseThrow(() -> new NoSuchElementException("해당 옵션을 찾을 수 없습니다 id :  " + optionId));
+
+        checkOptionInGift(option, giftId);
+        checkDuplicateOptionName(gift, option.getName());
+
+        option.modify(optionRequest.getName(), optionRequest.getQuantity());
+
+    }
+
+    @Transactional
+    public void subtractOptionToGift(Long giftId, Long optionId, int quantity) {
+        Gift gift = giftRepository.findById(giftId)
+                .orElseThrow(() -> new NoSuchElementException("해당 상품을 찾을 수 없습니다 id :  " + giftId));
+        Option option = optionRepository.findById(optionId)
+                .orElseThrow(() -> new NoSuchElementException("해당 옵션을 찾을 수 없습니다 id :  " + optionId));
+
+        checkOptionInGift(option, giftId);
+
+        option.subtract(quantity);
+        optionRepository.save(option);
+    }
+
+    @Transactional
+    public void deleteOptionFromGift(Long giftId, Long optionId) {
+        Gift gift = giftRepository.findById(giftId)
+                .orElseThrow(() -> new NoSuchElementException("해당 상품을 찾을 수 없습니다 id :  " + giftId));
+        Option option = optionRepository.findById(optionId)
+                .orElseThrow(() -> new NoSuchElementException("해당 옵션을 찾을 수 없습니다 id :  " + optionId));
+
+        checkOptionInGift(option, giftId);
+
+        gift.removeOption(option);
+        giftRepository.save(gift);
+        optionRepository.delete(option);
+    }
+
+    public void checkOptionInGift(Option option, Long giftId) {
+        if (!option.getGift().getId().equals(giftId)) {
```

Comment:
gift 객체를 파리미터로 넘겨주고 다음과 같이 처리하면 보다 객체지향적인 코드를 작성할 수 있을 것 같습니다ㅎㅎ
```suggestion
        if (!gift.hasOption(optionId)) {
```

Reply:
코드가 훨씬 깔끔해지네요 감사합니다.